lib/assertions.ts

```diff
@@ -520,15 +520,15 @@ export default class DOMAssertions {
    * expression.
    *
    * @param {string} name
-   * @param {string|RegExp} value
+   * @param {RegExp|any} value
    * @param {string?} message
    *
    * @example
    * assert.dom('input.password-input').hasProperty('type', 'password');
    *
    * @see {@link #doesNotHaveProperty}
    */
-  hasProperty(name: string, value: string | RegExp, message?: string): DOMAssertions {
+  hasProperty(name: string, value: unknown, message?: string): DOMAssertions {
```

Comment:
why are you using `unknown` here instead of `RegExp | any` as specified above?

Reply:
`RegExp | any` doesn’t make any sense in typescript, it will still evaluate as `any`. And `unknown` is safer then `any` because it doesn’t imply any knowledge about type.

As for JSDoc annotation there’s no `unknown` type, so `any` is the closest alternative. And it’s still helpful to mention `RegExp` because it has special treatment (match instead of just equality check).

Reply:
@wuron is correct: types in TS are *sets*, and since `any` is already a superset of everything else, having `T | any`, for *any* type, is always exactly identical to just having `any`.

Reply:
Addendum: you can actually use `unknown` in JSDoc and *most* tools will just show "unknown" there, and anything TS-aware will actually correctly interpret it.

Reply:
> you can actually use unknown in JSDoc and most tools will just show "unknown" there, and anything TS-aware will actually correctly interpret it.

If that’s preferred I can set it to `RegExp | unknown` in JSDoc. But does it really change anything? JSDoc definitions are only used as documentation: to generate `API.md` file, or to hint types for non-TS environments. In both cases we want to communicate that this method accepts any value, hence `any` works just fine.

In TS definitions we shouldn’t use `any`, because it’s compatible with all types and allows us to do something like this without any warnings:
```ts
function hasProperty(param: any) {
  param.callRandomMethod();
  const a: boolean = param;
  const b: string = param;
}
```
Parameter with type `unknown` is much safer, and it still allows to pass any value.